controllers/azurecluster_reconciler.go

```diff
@@ -300,7 +300,10 @@ func (r *azureClusterReconciler) deleteNSG(ctx context.Context) error {
 
 // CreateOrUpdateNetworkAPIServerIP creates or updates public ip name and dns name
 func (r *azureClusterReconciler) createOrUpdateNetworkAPIServerIP() error {
-	if r.scope.Network().APIServerIP.Name == "" {
+	// Take user provided IP address, or generate if no name exists.
+	if r.scope.AzureCluster.Spec.NetworkSpec.PublicIP != nil && r.scope.AzureCluster.Spec.NetworkSpec.PublicIP.Name != "" {
```

Comment:
@alexeldeib one thing I'm thinking: is it necessary to pass in the unmanaged public IP to the list of public IP specs to reconcile? Since it's not mutable, it might make sense to just exclude it from the list if the condition above is true in `PublicIPSpecs()`, WDYT?

Reply:
I think this makes sense. 

Another thing I just noticed, though -- I think it's technically possible to have a DNS name that wouldn't match what we expect if we construct it using GenerateFQDN. So we may still want to fetch the FQDN from the real ip?